code-challenges.test.js

```diff
@@ -87,8 +196,45 @@ var vowelTester2 = "academy"
 var vowelTester3 = "challenge"
 // Expected output: 2
 
-
-
-
+describe("When vowelIndexFinder is called", () => {
+  it("return 1 if learn is called",() => {
+    //arrange
+    const argument = vowelTester1
+    //act
+    const actualResult = vowelIndexFinder(argument)
+    //assert
+  expect(actualResult).toEqual(1)
+})
+  it("return 0 if academy is called",() => {
+    //arrange
+    const argument = vowelTester2
+    //act
+    const actualResult = vowelIndexFinder(argument)
+    //assert
+  expect(actualResult).toEqual(0)
+  })
+  it("return 2 if challenge is called",() => {
+    //arrange
+    const argument = vowelTester3
+    //act
+    const actualResult = vowelIndexFinder(argument)
+    //assert
+  expect(actualResult).toEqual(2)
+  })
+})
 
 // b) Create the function that makes the test pass.
+
+const vowelIndexFinder = (string) => {
+  var i = 0;
+  var vowelIndex
+  var letterArray = string.split("")
+  while (i < letterArray.length) {
```

Comment:
interesting use of while-loop! This is a great optimization because the code will stop iterating through the array as soon as it encounters the first index. You may be able to simplify the code by just returning `i` without needing to use `vowelIndex` since they represent the same thing. also there's a keyword `break` that can be used in loops that short circuits the iteration